docs/site/Controllers.md

```diff
@@ -58,35 +64,54 @@ const spec = {
     '200': {
       description: 'greeting text',
       schema: {type: 'string'},
-    }
-  }
+    },
+  },
 };
 ```
 
-There are several ways to define `Routes` to Controller methods. The first example defines a route to the Controller without any magic.
+There are several ways to define `Routes` to Controller methods. The first
+example defines a route to the Controller without any magic.
 
 ```js
-app.route('get', '/greet', spec, MyController, 'greet');
+// ... in your application constructor
+this.route('get', '/greet', spec, MyController, 'greet');
 ```
 
-Decorators allow you to annotate your Controller methods with routing metadata, so LoopBack can call the `app.route()` function for you.
+Decorators allow you to annotate your Controller methods with routing metadata,
+so LoopBack can call the `app.route()` function for you.
 
 ```js
+import {get} from '@loopback/rest';
+
 class MyController {
   @get('/greet', spec)
-  greet(name) {
+  greet(name: string) {
+    return `hello ${name}`;
   }
 }
 
-app.controller(MyController);
+// ... in your application constructor
+this.controller(MyController);
 ```
 
 ## Specifying Controller APIs
 
-For larger LoopBack applications, you can organize your routes into API Specifications using the OpenAPI specification. The `@api` decorator takes a spec with type `ControllerSpec` which comprises of a string `basePath` and a [Paths Object](https://github.com/OAI/OpenAPI-Specification/blob/0e51e2a1b2d668f434e44e5818a0cdad1be090b4/versions/2.0.md#paths-object). It is _not_ a full [Swagger](https://github.com/OAI/OpenAPI-Specification/blob/0e51e2a1b2d668f434e44e5818a0cdad1be090b4/versions/2.0.md#swagger-object) specification.
+For larger LoopBack applications, you can organize your routes into API
+Specifications using the OpenAPI specification. The `@api` decorator takes a
+spec with type `ControllerSpec` which comprises of a string `basePath` and a
+[Paths Object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#paths-object)
+Note that it is _not_ a full
+[OpenAPI](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#oasObject)
+specification.
 
 ```js
-app.api({
+// ... in your application constructor
```

Comment:
This won't be in the Application Constructor! Ideally it's in the related controller file! And hence, `app` should be injected and the code below should be using `app` instead of `this`.